File: 프로그래머스/unrated/181932. 코드 처리하기/코드 처리하기.py
def solution(code):
    ret = ''
    mode = 0
    for idx, i in enumerate(code):
        if i == '1' and mode == 0:
            mode = 1
        elif i == '1' and mode == 1:
            mode = 0
        else:
            if mode == 0 and idx%2 == 0:
                ret += i
            elif mode == 1 and idx%2 == 1:
                ret += i
    if ret:
        return ret
    else:
        return 'EMPTY'

File: 프로그래머스/0/181932. 코드 처리하기/코드 처리하기.java
class Solution {
    public String solution(String code) {
        char[] arr = code.toCharArray();
		boolean mode = false;
		String answer = "";

		for (int i = 0; i < arr.length; i++) {
			if (!mode) {
				if (arr[i] == '1') {
					mode = true;
				} else {
					answer += i % 2 == 0 ? arr[i] : "";
				}
			} else {
				if (arr[i] == '1') {
					mode = false;
				} else {
					answer += i % 2 == 1 ? arr[i] : "";
				}
			}
		}
		return answer == "" ? "EMPTY" : answer;
    }
}